activerecord/lib/active_record/railties/databases.rake

```diff
@@ -94,24 +94,43 @@ db_namespace = namespace :db do
 
   # IMPORTANT: This task won't dump the schema if ActiveRecord::Base.dump_schema_after_migration is set to false
   task :_dump do
-    if ActiveRecord::Base.dump_schema_after_migration
-      case ActiveRecord::Base.schema_format
-      when :ruby then db_namespace["schema:dump"].invoke
-      when :sql  then db_namespace["structure:dump"].invoke
-      else
-        raise "unknown schema format #{ActiveRecord::Base.schema_format}"
+    dump_all = ActiveRecord::Base.dump_schema_after_migration
+    database_configs = ActiveRecord::DatabaseConfigurations.new(databases).configs_for(env_name: Rails.env)
+
+    if database_configs.count == 1
+      if dump_all && database_configs.first.schema_dump
+        case ActiveRecord::Base.schema_format
+        when :ruby then db_namespace["schema:dump"].invoke
+        when :sql  then db_namespace["structure:dump"].invoke
+        else
+          raise "unknown schema format #{ActiveRecord::Base.schema_format}"
+        end
+      end
+    else
+      ActiveRecord::Tasks::DatabaseTasks.for_each(databases) do |name, db_config|
```

Comment:
This needs to be using `database_configs.each` not `for_each` since `db_config` is not fully qualified.